cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -98,15 +98,29 @@ func (c *CStorPoolController) syncHandler(key string, operation common.QueueOper
 
 // cStorPoolEventHandler is to handle cstor pool related events.
 func (c *CStorPoolController) cStorPoolEventHandler(operation common.QueueOperation, cStorPoolGot *apis.CStorPool) (string, error) {
+	var zpoolDumpErr error
 	pool.RunnerVar = util.RealRunner{}
+	uid := string(cStorPoolGot.GetUID())
+	csp := pool.CStorPools[uid]
 	switch operation {
 	case common.QOpAdd:
 		glog.Infof("Processing cStorPool added event: %v, %v", cStorPoolGot.ObjectMeta.Name, string(cStorPoolGot.GetUID()))
 
 		// lock is to synchronize pool and volumereplica. Until certain pool related
 		// operations are over, the volumereplica threads will be held.
 		common.SyncResources.Mux.Lock()
+		if csp != nil {
```

Comment:
Is this possible to have this new block of code within an anonymous function.
Why? So that we can defer `common.SyncResources.Mux.Unlock()` at the start of this anonymous func.

refer - https://github.com/kubernetes/sample-controller/blob/master/controller.go#L193

Reply:
will make it as different function